docs/adrs/0078-use-aws-appconfig-feature-flags.md

```diff
@@ -0,0 +1,85 @@
+* Be respectful of budgetary concerns
+* Improve our ability to deliver code
+* Solve all of the problem scenarios below
+* Minimize impact on the already tight Milestone 1 schedule
+
+
+Problem Statement
+
+The MilMove project has multiple teams pushing code to shared repositories. All teams are currently using the same branch in each repo: main. This is very efficient pre-production, because all teams see each others' changes quickly. When we go to production (and maybe even a little before), this breaks down. A couple of example scenarios will illustrate this:
+
+Scenario 1: The Bug vs. The Incomplete Feature
+
+While one of our teams is part-way through an epic that adds a new feature, a critical bug is reported. A fix for the bug is quickly found and committed. But if we deploy code from the main branch at that point, half of the new feature will also be included, since it was already committed. Releasing half of a feature to production is generally frowned upon.
+
+Scenario 2: Changing Authentication
+
+After we have been live for several months, the customer requires us to move from using login.gov to authenticate users to Okta. Writing and testing the code is no problem. But when we deploy it, every user now has to set up a new authentication account. Many of those users will be in the middle of moving to their next duty station when we spring this on them. Users are constantly moving, there will not be a break in the schedule where we can do the changeover.
+
+Scenario 3: Prime Freeze
+
+For 30 days before we go live, the Prime (HSA) requires a code freeze for testing. We do not want to stop deploying for 30 days, and we really don't want to stop development for 30 days. Feature flags would allow us to build (and maybe even deploy) code during the freeze and remain productive.
+
+## Considered Alternatives
+
+* Option #0: Do Nothing. Stay the course, don't make any changes, figure out what to do when problems arise
+* Option #1: LaunchDarkly: This is the industry standard feature flag platform.
+* Option #2: Unleash: Open source, can be self-hosted
+* Option #3: AWS AppConfig: we're already using lots of AWS, what's a little more?
+
+## Decision Outcome
+
+Chosen Alternative: Option #3: AWS AppConfig
+
+LaunchDarkly is the industry standard, but there is no room for it in the budget. Unleash has similar features to AWS AppConfig, but requires more setup and maintenance effort when self-hosted. This leaves AWS AppConfig as the reasonable choice.
+
+
+## Pros and Cons of the Alternatives
+
+### Option #0: Do Nothing
+* '+' No change to App Eng workflow
+* '+' No additional setup for Platform Team
+* '-' We will at some point ship the wrong code
+* '-' Will break API versioning required by Prime
+
+
+### Option #1: LaunchDarkly
+* '+' Industry standard
+* '+' Feature Rich
+* '-' Expensive. No room in ODC budget
+* '-' Might not be allowed (It is FedRAMP moderate, not IL4)
```

Comment:
**question**: who could get us an answer on whether it is ok to use or not?

Reply:
If I was going to chase that down, I would start with the project ISSM. If they can't make the call, they know who can.